### DISCUSSION TITLE
xDrip+ sync follow

### DISCUSSION THREAD

**Discussion by PhilTU** [NightscoutFoundation/xDrip#3658] [General]
Conceptual question - if a device is running succesfully with xDrip+ sync follow and you select a different data source, specifically companion app, but that data source doesn't receive any data (as there is no companion app sending notifications), would it be expected that the xDrip+ sync continues to receive and display data ?

**Comment by Navid200** [comment]
No, that has never been the intent.

xDrip should never give you readings from a source other than the one you have chosen.  

The companion app is a very special concept though.  Theoretically, you can have a Dexcom sensor on one arm and Libre on the other and have the corresponding apps installed.  If both apps are included in the list of our supported companion apps, xDrip could show readings from both.  
In that case, it is the responsibility of the user to ensure that only one of the two apps is set to show notifications.
When I brought this up with the developers, I was told that was an edge case and will not happen under normal circumstances.
That is not what you asked.  But, it is the only scenario I can think of where you could get readings from an unintended source.

**Reply by PhilTU** [reply]
Thanks, this confirms my expectation and indeed that of the person observing the phenomenon.

**Reply by PhilTU** [reply]
I tried to reproduce using the latest nightly on two phones and failed. I got the message in the log file "Received remote BG packet but we are not set as a follower" when I changed the xDrip+ sync follower to have Companion App as its hardware data source. As one might expect.

**Reply by Navid200** [reply]
I don't think xDrip only getting readings from one data source is a problem/ bug.  Do you?

**Reply by PhilTU** [reply]
I don't. But I was in contact with someone on Reddit who had apparently observed this happening when they had three follower devices receiving data despite being changed to follower. I didn't know if it might be an unintended consequence of recent changes and have invited them to shar ethe information here.

From what I can see there isn't a problem, but I wasn't the one who experienced it initially.

**Reply by Navid200** [reply]
Well, they have posted and I am trying to help them understand.
But, I am not sure if they follow.

**Reply by PhilTU** [reply]
Continued in https://github.com/NightscoutFoundation/xDrip/discussions/3661
